### PR TITLE
Fix probe helper client listener and logging

### PR DESCRIPTION
**Pull request by tommyreddad**
The probe helper is facing issues where there is some mixup between the forward client and receiver client. The relevant options are mismatched currently.

## Proposed Changes
- Use the correct options for the forward client and receiver client.
- Improved logging to reduce clutter and be more informative.
- ACK received events even if they generate an error.

### CODE REVIEW

File: test/test_images/probe_helper/probe/handlers/cloudpubsub.go [hunk 1]
```diff
@@ -70,6 +72,7 @@ func (p *CloudPubSubSourceProbe) Forward(ctx context.Context, event cloudevents.
 		return fmt.Errorf("CloudPubSubSource probe event has no '%s' extension", topicExtension)
 	}
 	ctx = cecontext.WithTopic(ctx, fmt.Sprint(topic))
+	logging.FromContext(ctx).Infow("Publishing message to pubsub topic", zap.String("topic", fmt.Sprint(topic)))
```
Comment (zhongduo) [inline test/test_images/probe_helper/probe/handlers/cloudpubsub.go:75]: I don't think you need `Sprint`, did you try not having it? Worst case you can do `topic.String()` if that does not work.

**Comment by eclipselu** [comment]
Apart from the nits that Jimmy brought up.
/lgtm

**Comment by eclipselu** [comment]
oops sorry for the inconvenience. forgot to hold the PR.

File: test/test_images/probe_helper/probe/handlers/cloudpubsub.go (hunk 1 above)
Comment (tommyreddad) [reply test/test_images/probe_helper/probe/handlers/cloudpubsub.go:75]: `topic` is of type `interface{}`, it does not implement the `String` method. So `fmt.Sprint` just performs that cast. But I should probably have used `topic.(string)` instead, I will keep this in mind in the future.

**Pull request merged**